test/extended/testdata/test-cli-debug.yaml

```diff
@@ -0,0 +1,69 @@
+kind: List
+apiVersion: v1
+items:
+- kind: ImageStream
+  apiVersion: v1
+  metadata:
+    name: busybox
+  spec:
+    tags:
+    - name: latest
+      from:
+        name: busybox
+        kind: DockerImage
+
+- kind: DeploymentConfig
+  apiVersion: v1
+  metadata:
+    name: busybox1
+  spec:
+    replicas: 0
+    selector:
+      deploymentconfig: busybox1
+    template:
+      metadata:
+        labels:
+          deploymentconfig: busybox1
+      spec:
+        containers:
+        - name: busybox
```

Comment:
@jim-minter container name should be an empty string here, to ensure the deployment uses the image from the imagechangetrigger.

that might be why your test is passing despite @smarterclayton's observation that the wrong resource is being used.

my bad for not catching that sooner.